odl/deform/linearized_deform.py

```diff
@@ -0,0 +1,546 @@
+
+        self.displacement = displacement
+
+    def _call(self, template):
+        """Implementation of ``self(template)``.
+
+        Parameters
+        ----------
+        template : `domain` element
+            Given template that is to be deformed by the fixed
+            displacement field.
+        """
+        return linear_deform(template, self.displacement)
+
+    @property
+    def adjoint(self):
+        """Adjoint of the operator.
+
+        Returns
+        -------
+        adjoint: `LinDeformFixedTemplDerivAdj`
+            The adjoint of the operator.
+        """
+        return LinDeformFixedDispAdj(self.displacement)
+
+
+class LinDeformFixedDispAdj(Operator):
+
+    """Adjoint of the fixed displacement linearized deformation operator.
+
+    This operator computes the adjoint of the linear operator that
+    map a template ``I`` to its deformation using a fixed displacement ``v``:
+
+        I --> exp(-div(v)) * I(Id - v)
+
+    Here, the ``I`` is the given template, ``Id`` is the identity mapping:
+
+        y --> y
+
+    the vector field ``v`` is a displacement field:
+
+        y --> v(y)
+
+    and the ``y`` is an element in the domain.
+
+    Here, ``exp(-div(v))`` is an approximation of the determinant of the
+    Jacobian of ``(Id + v)^{-1}``, which is accurate in some sense if the
+    magnitude of``v`` is close to ``0``.
+    """
+
+    def __init__(self, displacement):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        displacement : `ProductSpace` element
+            Fixed displacement field used in the linearized deformation.
+
+        Examples
+        --------
+        >>> import odl
+        >>> space = odl.uniform_discr(0, 1, 5)
+        >>> disp_field_space = odl.ProductSpace(space, space.ndim)
+        >>> disp_field = disp_field_space.element([[0.2, 0.2, 0.2, 0.2, 0.2]])
+        >>> template = space.element([0, 0, 1, 0, 0])
+        >>> op = LinDeformFixedDispAdj(disp_field)
+        >>> op(template)
+        uniform_discr(0.0, 1.0, 5).element([0.0, 0.0, 0.0, 1.0, 0.0])
+        """
+        if not isinstance(displacement.space, ProductSpace):
+            raise TypeError('`displacement` {!r} not an element'
+                            'of `ProductSpace`'.format(displacement))
+
+        domain_space = displacement[0].space
+        range_space = domain_space
+
+        Operator.__init__(self, domain_space, range_space, linear=True)
+
+        self.displacement = displacement
+
+    def _call(self, template):
+        """Implement ``self(template)```.
+
+        Parameters
+        ----------
+        template : `domain` element
+            Given template that is to be deformed by the fixed
+            displacement field.
+        """
+        div_op = Divergence(range=template.space, method='forward',
+                            padding_method='symmetric')
+        jacobian_det = np.exp(-div_op(self.displacement))
```

Comment:
We should keep an eye on that one, too. Perhaps we want to cache it. Add a `TODO` comment so we remember later.